tests/returntypes/transforms/test_split_non_commuting_new.py

```diff
@@ -0,0 +1,340 @@
+        split, _ = split_non_commuting(qs)
+        assert len(split) == expected
+
+    @pytest.mark.parametrize(
+        "tape,group_coeffs",
+        [(non_commuting_tape2, [[0, 2], [1, 3]]), (non_commuting_tape3, [[0, 3], [1, 4], [2, 5]])],
+    )
+    def test_non_commuting_group_right_reorder(self, tape, group_coeffs):
+        """Test that the output is of the correct order"""
+        split, fn = split_non_commuting(tape)
+        assert all(np.array(fn(group_coeffs)) == np.arange(len(split) * 2))
+
+        qs = qml.tape.QuantumScript(tape.operations, tape.measurements)
+        split, fn = split_non_commuting(qs)
+        assert all(np.array(fn(group_coeffs)) == np.arange(len(split) * 2))
+
+    @pytest.mark.parametrize("meas_type", obs_fn)
+    def test_different_measurement_types(self, meas_type):
+        """Test that expval, var and sample are correctly reproduced"""
+        with qml.tape.QuantumTape() as tape:
+            qml.PauliZ(0)
+            qml.Hadamard(0)
+            qml.CNOT((0, 1))
+            meas_type(qml.PauliZ(0) @ qml.PauliZ(1))
+            meas_type(qml.PauliX(0) @ qml.PauliX(1))
+            meas_type(qml.PauliZ(0))
+            meas_type(qml.PauliX(0))
+        the_return_type = tape.measurements[0].return_type
+        split, _ = split_non_commuting(tape)
+        for new_tape in split:
+            for meas in new_tape.measurements:
+                assert meas.return_type == the_return_type
+
+        qs = qml.tape.QuantumScript(tape.operations, tape.measurements)
+        split, _ = split_non_commuting(qs)
+        for new_tape in split:
+            for meas in new_tape.measurements:
+                assert meas.return_type == the_return_type
+
+    def test_raise_not_supported(self):
+        """Test that NotImplementedError is raised when probabilities or samples are called"""
+        with qml.tape.QuantumTape() as tape:
+            qml.expval(qml.PauliZ(0))
+            qml.probs(wires=0)
+
+        with pytest.raises(NotImplementedError, match="non-commuting observables are used"):
+            split_non_commuting(tape)
+
+
+class TestIntegration:
+    """Integration tests for ``qml.transforms.split_non_commuting()``"""
+
+    def test_expval_non_commuting_observables(self):
+        """Test expval with multiple non-commuting operators"""
+        dev = qml.device("default.qubit", wires=6)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Hadamard(1)
+            qml.Hadamard(0)
+            qml.PauliZ(0)
+            qml.Hadamard(3)
+            qml.Hadamard(5)
+            qml.T(5)
+            return [
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+                qml.expval(qml.PauliX(0)),
+                qml.expval(qml.PauliZ(1)),
+                qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
+                qml.expval(qml.PauliX(3)),
+                qml.expval(qml.PauliY(5)),
+            ]
+
+        res = circuit()
```

Comment:
It might worth to check that it is a tuple at the end with arrays of expected shape.

Reply:
Added tests for the structure of the tuple.